modular_citadel/code/modules/reagents/reagent container/hypovial.dm

```diff
@@ -29,6 +29,19 @@
 /obj/item/reagent_containers/glass/bottle/vial/on_reagent_change()
 	update_icon()
 
+/obj/item/reagent_containers/glass/bottle/vial/reskin_obj(mob/M) //Makes the vials completely reskinnable, and renames them - overrides /obj/proc/reskin_obj
```

Comment:
Why do we need a hard override for this?

Reply:
Because obj's reskin_obj doesn't provide the exact functionality I need. I could have altered it in obj (and it's not like it would change much) but for the sake of not fucking with the parent class to literally every game object ever, I decided to override it.

Reply:
what exactly do you 'need' that this is required by? It works just fine as is?

Reply:
Only once. If you read the comment on the function declaration, it's there solely to make the vials recolourable as needed.

Because someone else made the executive decision that this variant of a bottle needed to have multiple colour states, and these aren't unique items like the barkeep's gun or the captain's pen.